File: SWEA D2/d2_12712.java
// 12712 파리퇴치3
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
import java.io.IOException;

public class d2_12712 {
  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringBuilder sb = new StringBuilder();

    int T = Integer.parseInt(br.readLine());
    for (int t = 1; t <= T; t++) {
      // 입력
      StringTokenizer st = new StringTokenizer(br.readLine(), " ");
      int N = Integer.parseInt(st.nextToken());
      int M = Integer.parseInt(st.nextToken());
      int[][] mat = new int[N][N];
      for (int i = 0; i < N; i++) {
        st = new StringTokenizer(br.readLine(), " ");
        for (int j = 0; j < N; j++) {
          mat[i][j] = Integer.parseInt(st.nextToken());
        }
      }

      // Logic
      int max_count = 0;
      int count_t, count_x;
      for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
          if (i == 0 && (j == 0 || j == N - 1) ||
              i == N - 1 && (j == 0 || j == N - 1)) {
            continue;
          }
          count_t = mat[i][j];
          count_x = mat[i][j];
          for (int k = 1; k < M; k++) {
            if (j - k >= 0)
              count_t += mat[i][j - k];  // + 아래
            if (j + k < N)
              count_t += mat[i][j + k];  // + 위

            if (i - k >= 0) {
              count_t += mat[i - k][j];  // + 왼쪽
              if (j - k >= 0)
                count_x += mat[i - k][j - k];  // x 왼쪽 아래
              if (j + k < N)
                count_x += mat[i - k][j + k];  // x 왼쪽 위
            }

            if (i + k < N) {
              count_t += mat[i + k][j];  // + 오른쪽
              if (j - k >= 0)
                count_x += mat[i + k][j - k];  // x 오른쪽 아래
              if (j + k < N)
                count_x += mat[i + k][j + k];  // x 오른쪽 위
            }
          }
          max_count = Math.max(max_count, count_t);
          max_count = Math.max(max_count, count_x);
        }
      }

      // 출력
      sb.append("#").append(t).append(" ").append(max_count).append("\n");
    }

    System.out.println(sb.toString());
  }
}

File: SWEA D2/d2_12712.py
# 12712 파리퇴치3
from collections import deque

T = int(input())
for t in range(T):
    N, M = map(int, input().split())
    matrix = [list(map(int, input().split())) for _ in range(N)]

    max_count = 0
    for x in range(N):
        for y in range(N):
            # 스프레이 노즐 + 형태
            count_t = matrix[x][y]
            for m in range(1, M):
                if 0 <= x + m < N:
                    count_t += matrix[x + m][y]
                if 0 <= x - m < N:
                    count_t += matrix[x - m][y]
                if 0 <= y + m < N:
                    count_t += matrix[x][y + m]
                if 0 <= y - m < N:
                    count_t += matrix[x][y - m]
            # 스프레이 노즐 x 형태
            count_x = matrix[x][y]
            for m in range(1, M):
                if 0 <= x + m < N and 0 <= y + m < N:
                    count_x += matrix[x + m][y + m]
                if 0 <= x + m < N and 0 <= y - m < N:
                    count_x += matrix[x + m][y - m]
                if 0 <= x - m < N and 0 <= y + m < N:
                    count_x += matrix[x - m][y + m]
                if 0 <= x - m < N and 0 <= y - m < N:
                    count_x += matrix[x - m][y - m]
            # 최대값 계산
            max_count = max(max_count, count_t, count_x)
    print(f"#{t + 1}", max_count)